lib/devicetrust/authn/authn.go

```diff
@@ -60,14 +60,56 @@ func NewCeremony() *Ceremony {
 //
 // The outcome of the authentication ceremony is a pair of user certificates
 // augmented with device extensions.
-func (c *Ceremony) Run(ctx context.Context, devicesClient devicepb.DeviceTrustServiceClient, certs *devicepb.UserCertificates) (*devicepb.UserCertificates, error) {
+func (c *Ceremony) Run(
+	ctx context.Context,
+	devicesClient devicepb.DeviceTrustServiceClient,
+	certs *devicepb.UserCertificates,
+) (*devicepb.UserCertificates, error) {
 	switch {
 	case devicesClient == nil:
 		return nil, trace.BadParameter("devicesClient required")
 	case certs == nil:
 		return nil, trace.BadParameter("certs required")
 	}
 
+	newCerts, err := c.run(ctx, devicesClient, &devicepb.AuthenticateDeviceInit{
+		UserCertificates: &devicepb.UserCertificates{
+			// Forward only the SSH certificate, the TLS identity is part of the
+			// connection.
+			SshAuthorizedKey: certs.SshAuthorizedKey,
+		},
+	})
+	return newCerts, trace.Wrap(err)
+}
+
+// RunWeb performs on-behalf-of device authentication. It exchanges a webToken
+// issued for the Web UI for a device authentication attempt.
+//
+// On success the WebSession certificates, for the session associated to
+// webToken, are augmented with device extensions.
+func (c *Ceremony) RunWeb(
+	ctx context.Context,
+	devicesClient devicepb.DeviceTrustServiceClient,
+	webToken *devicepb.DeviceWebToken,
+) error {
+	switch {
+	case devicesClient == nil:
+		return trace.BadParameter("devicesClient required")
+	case webToken == nil:
+		return trace.BadParameter("webToken required")
+	}
```

Comment:
I'd rather use 2 ifs here

Reply:
Ack. Stylistically this is fine in Go; some prefer if/else chains, some prefer switches.